pkg/devfile/parser/parse_test.go

```diff
@@ -2389,8 +2392,10 @@ func Test_parseParentAndPlugin_RecursivelyReference_withMultipleURI(t *testing.T
 	testServer3.Start()
 	defer testServer3.Close()
 	t.Run("it should error out if URI is recursively referenced with multiple references", func(t *testing.T) {
-		err := parseParentAndPlugin(devFileObj)
-		expectedErr := fmt.Sprintf("URI %v%v is recursively referenced", httpPrefix, uri1)
+		err := parseParentAndPlugin(devFileObj, &resolutionContextTree{}, resolverTools{})
+		// devfile has an cycle in references: main devfile -> uri: http://127.0.0.1:8080 -> uri: http://127.0.0.1:9090 -> uri: http://127.0.0.1:8090 -> uri: http://127.0.0.1:8080
```

Comment:
I am thinking out aloud, can we have one test case where its a mixture?

uri->kube crd->id->uri

Reply:
Updated this existing test, to have 3 different types of import reference

Reply:
thx but the commment was not updated

Reply:
oops, updated